src/collectors/kafka_consumer_lag/kafka_consumer_lag.py

```diff
@@ -103,48 +80,97 @@ def collect(self):
-                    cmd = [
-                        'kafka.tools.ConsumerOffsetChecker',
-                        '--group',
-                        consumer_group,
-                        '--topic',
-                        topic,
-                        '--zookeeper',
-                        zookeeper
-                    ]
-
-                    raw_output = self.run_command(cmd)
-                    if raw_output is None:
-                        return
-
-                    total_lag = 0
-                    for i, output in enumerate(raw_output[0].split('\n')):
-                        if i == 0:
-                            continue
-
-                        items = output.strip().split(' ')
-                        metrics = [item for item in items if item]
-
-                        if not metrics:
-                            continue
-
-                        metric = ConsumerMetric(cluster_name, metrics)
-                        total_lag += metric.get_consumer_lag()
-                        consumer_lag_metric_name = metric.get_consumer_lag_metric_name()
-                        self.publish(consumer_lag_metric_name, metric.get_consumer_lag())
-
-                    total_lag_metric_name = self.get_total_lag_metric_name(cluster_name, consumer_group, topic)
-                    self.publish(total_lag_metric_name, total_lag)
-                except Exception as e:
-                    self.log.error(e)
+                    metric = ConsumerMetric(cluster_name, metrics)
+                    total_lag += metric.get_consumer_lag()
+                    consumer_lag_metric_name = metric.get_consumer_lag_metric_name()
+                    self.publish(consumer_lag_metric_name, metric.get_consumer_lag())
+                    if topic is None:
+                        topic = metric.topic
+
+                total_lag_metric_name = self.get_total_lag_metric_name(cluster_name, consumer_group, topic)
+                self.publish(total_lag_metric_name, total_lag)
+            except Exception as e:
+                self.log.error(e)
+
+    def get_command(self, consumer_group, zookeeper, is_0_9_above):
+        """
+        :param consumer_group:
+        :param zookeeper:
+        :param is_0_9_above:
+        :return: Command for 0.8.x or 0.9.x based on versions
+        """
+        if is_0_9_above:
+            cmd = [
+                '--group',
+                consumer_group,
+                '--zookeeper',
+                zookeeper,
+                '--describe'
+            ]
+        else:
+            cmd = [
+                'kafka.tools.ConsumerOffsetChecker',
+                '--group',
+                consumer_group,
+                '--zookeeper',
+                zookeeper
+            ]
+        return cmd
+
+    def get_consumer_groups(self, zookeeper, is_0_9_above):
+        """
+        Get consumer groups from config or kafka (0.9 above)
+        :return: Consumer groups
+        """
+        if is_0_9_above:
```

Comment:
also verify that `consumer_groups` is empty since this logic would override any groups passed in